core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1179,13 +1179,15 @@ public void run() {
       return delegate;
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public void start(Listener<RespT> observer, Metadata headers) {
       PickSubchannelArgs args = new PickSubchannelArgsImpl(method, headers, callOptions);
       InternalConfigSelector.Result result = configSelector.selectConfig(args);
       Status status = result.getStatus();
       if (!status.isOk()) {
         executeCloseObserverInContext(observer, status);
+        delegate = (ClientCall<ReqT, RespT>) NOOP_CALL;
```

Comment:
It would have been nice to avoid the cast and I think it is possible to achieve that but I understand this pattern was copied from `ClientInterceptors.NOOP_CALL` and `DelayedClientCall.NOOP_CALL`. If possible the refactoring can combine these into one and also avoid the cast?

Reply:
> If possible the refactoring can combine these into one and also avoid the cast? 

I think so.